src/components/new-suggestion/suggestion-page-section.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useEffect, useState } from "react"
+import { XMarkIcon } from "@heroicons/react/24/outline"
+import ProgressBar from "@/components/new-suggestion/progress-bar/progress-bar"
+import SongInformationArea from "@/components/new-suggestion/areas/song-information.area"
+import ReviewArea from "@/components/new-suggestion/areas/review.area"
+import { Area } from "@/constants/area"
+import { FormProvider, useForm } from "react-hook-form"
+import {
+  InputsSongInformation,
+  NewSuggestion,
+  NewSuggestionInstrument,
+} from "@/interfaces/new-suggestion"
+import InstrumentsArea from "@/components/new-suggestion/areas/instruments/instruments.area"
+import { Database } from "@/types/database"
+import { Instrument } from "@/types/database-types"
+import { useSupabaseClient } from "@supabase/auth-helpers-react"
+import { getInstruments } from "@/services/instrument.service"
+import Spinner from "@/components/utils/spinner"
+import ErrorPopup from "@/components/popups/error-popup"
+import { submitSongInformationForm } from "@/helpers/new-suggestion.helper"
+
+interface SuggestionPageSectionProps {
+  title: string
+  newSuggestion: NewSuggestion
+  startingArea: Area
+  onCloseClicked(): void
+  onSongInformationSubmit(songInformation: InputsSongInformation): void
+  onInstrumentSubmit(newInstruments: NewSuggestionInstrument[]): void
+  onAreaSelect(newArea: Area): void
+  onReviewSubmit(onSuccess: () => void, onError: () => void): void
+}
+
+const SuggestionPageSection = ({
+  title,
+  newSuggestion,
+  startingArea,
+  onReviewSubmit,
+  onAreaSelect,
+  onCloseClicked,
+  onInstrumentSubmit,
+  onSongInformationSubmit,
+}: SuggestionPageSectionProps) => {
+  const [activeArea, setActiveArea] = useState<Area>(startingArea)
+  const supabaseClient = useSupabaseClient<Database>()
+  const [showSpinner, setShowSpinner] = useState<boolean>(false)
+  const [showLoadingError, setShowLoadingError] = useState<boolean>(false)
+
+  const methods = useForm<InputsSongInformation>({
+    defaultValues: {
+      ...newSuggestion,
+      artist: newSuggestion.artist.join(","),
+    } as InputsSongInformation,
+    shouldFocusError: false,
+  })
+
+  const [instrumentList, setInstrumentList] = useState<Instrument[]>([])
+  const [newSuggestionInstruments, setNewSuggestionInstruments] = useState<
+    NewSuggestionInstrument[]
+  >(newSuggestion.instruments)
```

Comment:
formatting seems a bit awkward

Reply:
Agree, but Prettier does this to prevent it to go over the line notified in my editor. Does make it more readable in split screen editor mode
![image](https://github.com/ASVGay/the-rhapsodies/assets/4838158/7250f5d2-f08d-404c-bed8-54c666f53c74)